src/NHibernate/Linq/DefaultQueryProvider.cs

```diff
@@ -282,6 +282,9 @@ public virtual void SetResultTransformerAndAdditionalCriteria(IQuery query, NhLi
 
 		public int ExecuteDml<T>(QueryMode queryMode, Expression expression)
 		{
+			if (Collection != null)
+				throw new NotSupportedException("DML operations are not supported for filters.");
```

Comment:
Should probably be `InvalidOperationException`. However, I'm not sure.

Reply:
As potentially it could be implemented I think NotSupported/NotImplemented exceptions are more appropriate.

Reply:
NotSupported is fine